server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -400,6 +400,7 @@ private void updateDesireBalanceMetrics(
             routingAllocation.metadata(),
             routingAllocation.routingNodes(),
             routingAllocation.clusterInfo(),
+            () -> {},
```

Comment:
Could we make this into a static constant so that it has a name, something like `NEVER_CANCELLED` perhaps? Otherwise it leaves the reader wondering what this lambda is for (and also saves allocating a fresh lambda each time it's called, although the compiler might be clever enough to skip that anyway)